src/test/end-to-end/pages/solicitorApplyProbate/grantOfProbate/page3.js

```diff
@@ -1,12 +1,48 @@
+    } else {
+        await I.fillField('#primaryApplicantForenames', grantOfProbateConfig.page2_primaryApplicantFirstName);
+        await I.fillField('#primaryApplicantSurname', grantOfProbateConfig.page2_primaryApplicantSurname);
+        await I.click(`#primaryApplicantHasAlias-${grantOfProbateConfig.optionYes}`);
+        await I.fillField('#solsExecutorAliasNames', grantOfProbateConfig.page2_primaryApplicantWillName);
+        await I.click(`#primaryApplicantIsApplying-${grantOfProbateConfig.optionYes}`);
+
+        await I.click(grantOfProbateConfig.UKpostcodeLink);
+        await I.fillField('#primaryApplicantAddress_AddressLine1', grantOfProbateConfig.address_line1);
+        await I.fillField('#primaryApplicantAddress_AddressLine2', grantOfProbateConfig.address_line2);
+        await I.fillField('#primaryApplicantAddress_AddressLine3', grantOfProbateConfig.address_line3);
+        await I.fillField('#primaryApplicantAddress_PostTown', grantOfProbateConfig.address_town);
+        await I.fillField('#primaryApplicantAddress_County', grantOfProbateConfig.address_county);
+        await I.fillField('#primaryApplicantAddress_PostCode', grantOfProbateConfig.address_postcode);
+        await I.fillField('#primaryApplicantAddress_Country', grantOfProbateConfig.address_country);
+
+        await I.click(`#otherExecutorExists-${grantOfProbateConfig.optionYes}`);
+
+        await I.waitForText(grantOfProbateConfig.page2_waitForAdditionalExecutor, testConfig.TestTimeToWaitForText);
+
+        await I.click('#solsAdditionalExecutorList > div > button');
+        await I.fillField('#solsAdditionalExecutorList_0_additionalExecForenames', grantOfProbateConfig.page2_executorFirstName);
+        await I.fillField('#solsAdditionalExecutorList_0_additionalExecLastname', grantOfProbateConfig.page2_executorSurname);
+        await I.click(`#solsAdditionalExecutorList_0_additionalExecNameOnWill-${grantOfProbateConfig.optionYes}`);
+        await I.click(`#solsAdditionalExecutorList_0_additionalApplying-${grantOfProbateConfig.optionYes}`);
+
+        await I.waitForElement('#solsAdditionalExecutorList_0_additionalExecAddress_additionalExecAddress_postcodeInput', testConfig.TestTimeToWaitForText);
+        await I.fillField('#solsAdditionalExecutorList_0_additionalExecAddress_additionalExecAddress_postcodeInput', grantOfProbateConfig.page2_executorPostcode);
+        await I.click('#solsAdditionalExecutorList_0_additionalExecAddress_additionalExecAddress > div  > div > button');
+        await I.waitForElement('#solsAdditionalExecutorList_0_additionalExecAddress_additionalExecAddress_addressList', testConfig.TestTimeToWaitForText);
+        await I.fillField('#solsAdditionalExecutorList_0_additionalExecAliasNameOnWill', grantOfProbateConfig.page2_executorAliasName);
+        await I.retry(10).selectOption('#solsAdditionalExecutorList_0_additionalExecAddress_additionalExecAddress_addressList', grantOfProbateConfig.page2_executorAddress);
+    }
```

Comment:
COULD: retrying 10 times feels a bit implicit, wouldn't a change to waitForEnabled from waitForElement  #solsAdditionalExecutorList_0_additionalExecAddress_additionalExecAddress_addressList do the trick, perhaps with a wait for the option too? Happy if it works tho as the key thing is it doesn't cause unnecessary waits.